src/test/java/bio/overture/ego/controller/UserControllerTest.java

```diff
@@ -241,12 +254,333 @@ public void createUser_EmailAlreadyExists_Conflict() {
+
+    // setup incoming user with IDToken, same email
+    val idToken = new IDToken();
+    idToken.setFamilyName(lastName);
+    idToken.setGivenName(firstName);
+    idToken.setEmail(email);
+    idToken.setProviderType(providerType);
+    idToken.setProviderId(providerId);
+
+    //  Assert not found by providerInfo
+    assertTrue(userService.findByProviderTypeAndProviderId(providerType, providerId).isEmpty());
+
+    //  Assert found by email-as-providerid and default providerType
+    assertTrue(userService.findByProviderTypeAndProviderId(DEFAULT_PROVIDER, email).isPresent());
+    // create user from idToken
+    val userFromToken = userService.getUserByToken(idToken);
+
+    // assert user exists
+    assertEquals(userFromToken.getClass(), User.class);
+    initStringRequest().endpoint("users/%s", userFromToken.getId()).getAnd().assertOk();
+
+    //  assert existingUser equals userFromToken
+    assertEquals(existingUser.getId(), userFromToken.getId());
+
+    //  assert user properties match idToken properties
+    assertEquals(userFromToken.getFirstName(), idToken.getGivenName());
+    assertEquals(userFromToken.getLastName(), idToken.getFamilyName());
+    assertEquals(userFromToken.getProviderType(), idToken.getProviderType());
+    assertEquals(userFromToken.getProviderId(), idToken.getProviderId());
+    assertEquals(userFromToken.getEmail(), idToken.getEmail());
+
+    //    assert existingUser properties are updated from userFromToken
+    assertEquals(existingUser.getEmail(), userFromToken.getEmail());
+    assertEquals(existingUser.getFirstName(), userFromToken.getFirstName());
+    assertEquals(existingUser.getLastName(), userFromToken.getLastName());
+    assertEquals(existingUser.getCreatedAt(), userFromToken.getCreatedAt());
+    assertEquals(existingUser.getType(), userFromToken.getType());
+    assertEquals(existingUser.getStatus(), userFromToken.getStatus());
+    assertNotEquals(existingUser.getProviderType(), userFromToken.getProviderType());
+    assertNotEquals(existingUser.getProviderId(), userFromToken.getProviderId());
+  }
+
+  @Test
+  public void getUserByToken_NonExistingProviderInfoNonExistingEmail_CreateNewUserSuccess() {
+    val userName = entityGenerator.generateNonExistentUserName();
+    val firstName = userName.split(" ")[0];
+    val lastName = userName.split(" ")[1];
+    val email = format("%s@example.com", userName);
+    val providerType = FACEBOOK;
+    val providerId = generateNonExistentProviderId(userService);
+
+    val idToken = new IDToken();
+    idToken.setFamilyName(lastName);
+    idToken.setGivenName(firstName);
+    idToken.setEmail(email);
+    idToken.setProviderType(providerType);
+    idToken.setProviderId(providerId);
+
+    // Assert not found by providerInfo
+    assertTrue(userService.findByProviderTypeAndProviderId(providerType, providerId).isEmpty());
+    // Assert not found by email
+    assertTrue(userService.findByName(userName).isEmpty());
```

Comment:
add another assertion to ensure that email doesnt already exist

Reply:
in this case email and name are still the same, so `findByName` should satisfy that requirement? It will change once I start on getting rid of the `name` field.